<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Guangchuan Ji" />

<meta name="date" content="2018-06-24" />

<title>Heterozygosity and F_{st} for admixed populations</title>

<script src="data:application/x-javascript;base64,JChkb2N1bWVudCkucmVhZHkoZnVuY3Rpb24oKXsKICAgICQoJ1tkYXRhLXRvZ2dsZT0idG9vbHRpcCJdJykudG9vbHRpcCgpOwogICAgJCgnW2RhdGEtdG9nZ2xlPSJwb3BvdmVyIl0nKS5wb3BvdmVyKCk7Cn0pOwo="></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Heterozygosity and <span class="math inline">\(F_{st}\)</span> for admixed populations</h1>
<h4 class="author"><em>Guangchuan Ji</em></h4>
<h4 class="date"><em>2018-06-24</em></h4>



<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">echo =</span> <span class="ot">TRUE</span>, <span class="dt">comment =</span> <span class="ot">NA</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(knitr)
<span class="kw">library</span>(kableExtra)</code></pre></div>
<div id="function-to-calculate-the-heterozygosity-of-a-population-at-a-locus" class="section level2">
<h2>Function to calculate the heterozygosity of a population at a locus</h2>
<p>A diploid organism is heterozygous at a gene locus when its cells contain two different alleles of a gene. The cell or organism is called a heterozygote specifically for the allele in question, and therefore, heterozygosity refers to a specific genotype. High heterozygosity means lots of genetic variability. Low heterozygosity means little genetic variability.</p>
<p>We can use the expected heterozygosity to estimate the gene diversity. The expected heterozygosity in a population for a given locus with I distinct alleles is defined as <span class="math inline">\(H = 1 - \sum_{i=1}^I p_{i}^2\)</span>, where <span class="math inline">\(p_{i}\)</span> is the frequency of allele i.</p>
<p>To calculate the expected heterozygosity, we can use the code below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get.het &lt;-<span class="st"> </span><span class="cf">function</span>(allele.freq)
{
<span class="dv">1</span> <span class="op">-</span><span class="st"> </span><span class="kw">sum</span>(allele.freq<span class="op">^</span><span class="dv">2</span>)
}</code></pre></div>
<p>The input is the vector of allele ferquencies (should add up to 1).</p>
<div id="example" class="section level3">
<h3>Example</h3>
<p>If we have three allele frequencies: 0.2, 0.3, 0.5</p>
<p>First check the allele frequencies add up to 1:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(<span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>)</code></pre></div>
<pre><code>[1] 1</code></pre>
<p>And now we check the missing data NA:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">is.na</span>(<span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>))</code></pre></div>
<pre><code>[1] FALSE FALSE FALSE</code></pre>
<p>Then the expected heterozygosity is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get.het</span>(<span class="kw">c</span>(<span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>))</code></pre></div>
<pre><code>[1] 0.62</code></pre>
</div>
</div>
<div id="function-to-calculate-f_st-for-pop.1-and-pop.2" class="section level2">
<h2>Function to calculate <span class="math inline">\(F_{st}\)</span> for pop.1 and pop.2</h2>
<p>The “fixation index” <span class="math inline">\(F_{st}\)</span> is a measure of genetic divergence between an admixed population and its parental source populations. <span class="math inline">\(F_{st}\)</span> can be defined as <span class="math inline">\(F_{st}\)</span> = (<span class="math inline">\(H_t\)</span> - <span class="math inline">\(H_s\)</span>)/H, where <span class="math inline">\(H_t\)</span> is the expected heterozygosity of the overall population, and <span class="math inline">\(H_s\)</span> is the mean expected heterozygosity across subpopulations.</p>
<p>To calculate <span class="math inline">\(F_{st}\)</span>, we can use the codes below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get.Fst &lt;-<span class="st"> </span><span class="cf">function</span>(pop.<span class="dv">1</span>, pop.<span class="dv">2</span>)
{
  meta.pop &lt;-<span class="st"> </span>(pop.<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>pop.<span class="dv">2</span>)<span class="op">/</span><span class="dv">2</span>
  Ht &lt;-<span class="st"> </span><span class="kw">get.het</span>(meta.pop)
  Hs &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">c</span>(<span class="kw">get.het</span>(pop.<span class="dv">1</span>), <span class="kw">get.het</span>(pop.<span class="dv">2</span>)))
  Fst &lt;-<span class="st"> </span>(Ht <span class="op">-</span><span class="st"> </span>Hs)<span class="op">/</span>Ht
  Fst
}</code></pre></div>
<p>The input is pop.1, pop.2 (allele frequencies for the same locus for populations 1 and 2).</p>
<div id="example-1" class="section level3">
<h3>Example</h3>
<p>If we know the allele frequency of pop.1 is 0.3, and the allele frequency of pop.2 is 0.4.</p>
<p>First we check the missing data NA:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">is.na</span>(<span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.4</span>))</code></pre></div>
<pre><code>[1] FALSE FALSE</code></pre>
<p>Then we can get the <span class="math inline">\(F_{st}\)</span>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get.Fst</span>(<span class="fl">0.3</span>, <span class="fl">0.4</span>)</code></pre></div>
<pre><code>[1] 0.002849003</code></pre>
</div>
</div>
<div id="function-to-calculate-f_st-between-a-founder-population-and-the-admixed-population-for-the-k2-founder-population-case" class="section level2">
<h2>Function to calculate <span class="math inline">\(F_{st}\)</span> between a founder population and the admixed population for the K=2 founder population case</h2>
<p>In the following sections, we try to calculate the <span class="math inline">\(F_{st}\)</span> between a founder population and the admixed population.</p>
<p>First, we consider a special case in which only 2 fouding populations give rise to an admixed population. And we use <span class="math inline">\(\gamma_k\)</span> to present the admixture fraction corresponding to population k.</p>
<p>For the K = 2 founder population case, we can use the codes below to calculate <span class="math inline">\(F_{st}\)</span> between pop.1 and the admixed population:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get.Fst.admix &lt;-<span class="st"> </span><span class="cf">function</span>(pop.<span class="dv">1</span>, pop.<span class="dv">2</span>, gamma)
{
  pop.<span class="dv">3</span> &lt;-<span class="st"> </span>gamma<span class="op">*</span>pop.<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>gamma)<span class="op">*</span>pop.<span class="dv">2</span>
  meta.pop &lt;-<span class="st"> </span>(pop.<span class="dv">1</span> <span class="op">+</span><span class="st"> </span>pop.<span class="dv">3</span>)<span class="op">/</span><span class="dv">2</span>
  Ht &lt;-<span class="st"> </span><span class="kw">get.het</span>(meta.pop)
  Hs &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">c</span>(<span class="kw">get.het</span>(pop.<span class="dv">1</span>), <span class="kw">get.het</span>(pop.<span class="dv">3</span>)))
  Fst &lt;-<span class="st"> </span>(Ht <span class="op">-</span><span class="st"> </span>Hs)<span class="op">/</span>Ht
  Fst
}</code></pre></div>
<p>The input is pop.1, pop.2, gamma (gamma is the ancestry fraction corresponding to population 1).</p>
<div id="example-2" class="section level3">
<h3>Example</h3>
<p>Now we still consider the allele frequency of pop.1 is 0.3, and the allele frequency of pop.2 is 0.4, and fraction <span class="math inline">\(\gamma_k\)</span> = 0.25.</p>
<p>First we check the missing data NA:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">is.na</span>(<span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.25</span>))</code></pre></div>
<pre><code>[1] FALSE FALSE FALSE</code></pre>
<p>Then we can get <span class="math inline">\(F_{st}\)</span> between pop.1 and the admixed population:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get.Fst.admix</span>(<span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.25</span>)</code></pre></div>
<pre><code>[1] 0.001587022</code></pre>
</div>
</div>
<div id="function-to-calculate-f_st-for-pop.1-and-admixed-population-for-the-general-case-of-k-founding-population" class="section level2">
<h2>Function to calculate <span class="math inline">\(F_{st}\)</span> for pop.1 and admixed population for the general case of K founding population</h2>
<p>In this section, we consider the general case in which K founding populations give rise to an admixed population. Without loss of generality, we investigate <span class="math inline">\(F_{st}\)</span> between population 1 and the admixed population. And we use the codes below to get this <span class="math inline">\(F_{st}\)</span> value:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">get.Fst.admix.general &lt;-<span class="st"> </span><span class="cf">function</span>(pops, gamma)
{
  ##get admixed population
  admix.pop &lt;-<span class="st"> </span><span class="kw">t</span>(pops) <span class="op">%*%</span><span class="st"> </span>gamma
  
  ##get allele frequencies of &quot;meta-population&quot;
  meta.pop &lt;-<span class="st"> </span>(pops[<span class="dv">1</span>, ] <span class="op">+</span><span class="st"> </span>admix.pop)<span class="op">/</span><span class="dv">2</span>
  
  Ht &lt;-<span class="st"> </span><span class="kw">get.het</span>(meta.pop)
  Hs &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">c</span>(<span class="kw">get.het</span>(pops[<span class="dv">1</span>, ]), <span class="kw">get.het</span>(admix.pop)))
  Fst &lt;-<span class="st"> </span>(Ht <span class="op">-</span><span class="st"> </span>Hs)<span class="op">/</span>Ht
  Fst
}</code></pre></div>
<p>The input is a matrix with K rows and I columns (sums on rows is 1), vector of gammas (add up to 1).</p>
<div id="example-3" class="section level3">
<h3>Example</h3>
<p>We have the data in a matrix, where each row represents a founder population and each column an allele for the locus of interest.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pops &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">nrow =</span> <span class="dv">2</span>, <span class="dt">ncol =</span> <span class="dv">10</span>, <span class="kw">c</span>(<span class="fl">0.059</span>, <span class="fl">0.153</span>, <span class="fl">0.132</span>, <span class="fl">0.101</span>, <span class="fl">0.08</span>, <span class="fl">0.055</span>, <span class="fl">0.106</span>, <span class="fl">0.145</span>, <span class="fl">0.048</span>, <span class="fl">0.121</span>, <span class="fl">1.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>, <span class="fl">0.000</span>), <span class="dt">byrow=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>This is what the data looks like:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(pops)</code></pre></div>
<pre><code>[1] &quot;matrix&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rownames</span>(pops) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;pop1&quot;</span>, <span class="st">&quot;pop2&quot;</span>)
<span class="kw">colnames</span>(pops) &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;Allele&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(pops))
pops <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">kable</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">kable_styling</span>()</code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Allele 1
</th>
<th style="text-align:right;">
Allele 2
</th>
<th style="text-align:right;">
Allele 3
</th>
<th style="text-align:right;">
Allele 4
</th>
<th style="text-align:right;">
Allele 5
</th>
<th style="text-align:right;">
Allele 6
</th>
<th style="text-align:right;">
Allele 7
</th>
<th style="text-align:right;">
Allele 8
</th>
<th style="text-align:right;">
Allele 9
</th>
<th style="text-align:right;">
Allele 10
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
pop1
</td>
<td style="text-align:right;">
0.059
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
0.132
</td>
<td style="text-align:right;">
0.101
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:right;">
0.106
</td>
<td style="text-align:right;">
0.145
</td>
<td style="text-align:right;">
0.048
</td>
<td style="text-align:right;">
0.121
</td>
</tr>
<tr>
<td style="text-align:left;">
pop2
</td>
<td style="text-align:right;">
1.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
<td style="text-align:right;">
0.000
</td>
</tr>
</tbody>
</table>
<p>First check that the allele frequencies add up to 1:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rowSums</span>(pops)</code></pre></div>
<pre><code>pop1 pop2 
   1    1 </code></pre>
<p>Now also consider the following admixture proportions for each population:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gamma &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.7</span>)
<span class="kw">sum</span>(gamma)</code></pre></div>
<pre><code>[1] 1</code></pre>
<p>Then check the missing data NA:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">is.na</span>(pops)</code></pre></div>
<pre><code>     Allele 1 Allele 2 Allele 3 Allele 4 Allele 5 Allele 6 Allele 7
pop1    FALSE    FALSE    FALSE    FALSE    FALSE    FALSE    FALSE
pop2    FALSE    FALSE    FALSE    FALSE    FALSE    FALSE    FALSE
     Allele 8 Allele 9 Allele 10
pop1    FALSE    FALSE     FALSE
pop2    FALSE    FALSE     FALSE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">is.na</span>(gamma)</code></pre></div>
<pre><code>[1] FALSE FALSE</code></pre>
<p>Now calculate the <span class="math inline">\(F_{st}\)</span> between population 1 and the admixed population:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">get.Fst.admix.general</span>(pops, gamma)</code></pre></div>
<pre><code>[1] 0.1518406</code></pre>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
